pennylane_lightning/core/src/simulators/lightning_qubit/measurements/MeasurementsLQubit.hpp

```diff
@@ -580,11 +582,14 @@ class Measurements final
      */
     std::vector<std::size_t>
     generate_samples(const std::size_t num_samples,
-                     const std::optional<std::size_t> &seed = std::nullopt) {
+                     const std::optional<Measurements *> &m = std::nullopt) {
```

Comment:
Is it possible to not pass the entire measurement class into itself here?

Reply:
Duplicate of my comment above

Reply:
See reply above